protocol/x/listing/keeper/listing_test.go

```diff
@@ -0,0 +1,188 @@
+			duplicateMarket: false,
+			expectedErr:     types.ErrMarketNotFound,
+		},
+		"failure - duplicate market": {
+			ticker:          "TEST-USD",
+			duplicateMarket: true,
+			expectedErr:     nil,
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(
+			name, func(t *testing.T) {
+				mockIndexerEventManager := &mocks.IndexerEventManager{}
+				ctx, keeper, _, _, pricesKeeper, _, _, marketMapKeeper := keepertest.ListingKeepers(
+					t,
+					&mocks.BankKeeper{},
+					mockIndexerEventManager,
+				)
+
+				testMarketParams := pricestypes.MarketParam{
+					Pair:               "TEST-USD",
+					Exponent:           int32(-6),
+					ExchangeConfigJson: `{"test_config_placeholder":{}}`,
+					MinExchanges:       2,
+					MinPriceChangePpm:  uint32(800),
+				}
+
+				keepertest.CreateMarketsInMarketMapFromParams(
+					t,
+					ctx,
+					marketMapKeeper,
+					[]pricestypes.MarketParam{
+						testMarketParams,
+					},
+				)
+
+				marketId, err := keeper.CreateMarket(ctx, tc.ticker)
+				if tc.expectedErr != nil {
+					require.Error(t, err)
+				} else {
+					require.NoError(t, err)
+
+					// Check if the market was created
+					market, exists := pricesKeeper.GetMarketParam(ctx, marketId)
+					require.True(t, exists)
+					require.Equal(t, testMarketParams.Pair, market.Pair)
+					require.Equal(t, testMarketParams.Exponent, market.Exponent)
+					require.Equal(t, testMarketParams.MinExchanges, market.MinExchanges)
+					require.Equal(t, testMarketParams.MinPriceChangePpm, types.MinPriceChangePpm_LongTail)
+				}
+
+				if tc.duplicateMarket {
+					_, err = keeper.CreateMarket(ctx, tc.ticker)
+					require.ErrorContains(t, err, pricestypes.ErrMarketParamPairAlreadyExists.Error())
+				}
+			},
+		)
+	}
+}
+
+func TestCreatePerpetual(t *testing.T) {
+	tests := map[string]struct {
+		ticker         string
+		referencePrice uint64
+
+		expectedErr error
+	}{
+		"success": {
+			ticker:         "TEST-USD",
+			referencePrice: 1000000000,
+			expectedErr:    nil,
+		},
+		"failure - reference price 0": {
+			ticker:         "TEST-USD",
+			referencePrice: 0,
+			expectedErr:    types.ErrReferencePriceZero,
+		},
+		"failure - invalid market": {
+			ticker:      "INVALID-USD",
+			expectedErr: types.ErrMarketNotFound,
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(
+			name, func(t *testing.T) {
+				mockIndexerEventManager := &mocks.IndexerEventManager{}
+				ctx, keeper, _, _, pricesKeeper, perpetualsKeeper, _, marketMapKeeper := keepertest.ListingKeepers(
+					t,
+					&mocks.BankKeeper{},
+					mockIndexerEventManager,
+				)
+				keepertest.CreateLiquidityTiersAndNPerpetuals(t, ctx, perpetualsKeeper, pricesKeeper, 10)
+
+				// Create a marketmap with a single market
+				dydxMetadata, err := tickermetadata.MarshalDyDx(
+					tickermetadata.DyDx{
+						ReferencePrice: tc.referencePrice,
+						Liquidity:      0,
+						AggregateIDs:   nil,
+					},
+				)
+				require.NoError(t, err)
+
+				market := marketmaptypes.Market{
```

Comment:
non-blocking nit: There's some constants in another PR for market map that declares such constants, could look at moving these into a shared file.